rinja_parser/src/node.rs

```diff
@@ -568,17 +569,41 @@ impl<'a> Loop<'a> {
 pub struct Macro<'a> {
     pub ws1: Ws,
     pub name: &'a str,
-    pub args: Vec<&'a str>,
+    pub args: Vec<(&'a str, Option<WithSpan<'a, Expr<'a>>>)>,
     pub nodes: Vec<Node<'a>>,
     pub ws2: Ws,
 }
 
+macro_rules! check_duplicated_name {
+    ($names:ident, $arg_name:ident, $i:ident) => {
+        if !$names.insert($arg_name) {
+            return Err(nom::Err::Failure(ErrorContext::new(
+                format!("duplicated argument `{}`", $arg_name),
+                $i,
+            )));
+        }
+    };
+}
```

Comment:
Why didn't you write that as a function?

Reply:
I did originally, but forgot why it didn't work out. Gonna give it another try.